### PR TITLE
redesign course-navigation, add table of contents

### LINKED ISSUE #138: Content should be as high as course navigation
https://de.serlo.org/30195/uebersicht

@creitiv @Knorrke

### LINKED ISSUE #209: Course , dropdown menu
This menus are not easy to use. Especially when u have touchscreen. They disapeare when you put your finger away so you cant go to the section below.
 
course

### PR DESCRIPTION
**Pull request by Knorrke**
closes #138, closes #209

### CODE REVIEW

**Comment by inyono** [comment]
Can you please autoformat (i.e. Code > Reformat Code) `src/module/Entity/config/module.config.php` and `src/module/Entity/config/types.config.php`, so that there are no additional whitespace changes? Otherwise, we probably have merge conflicts in next PRs, too, like you had.

File: src/module/Entity/src/Entity/Entity/Entity.php [hunk 1]
```diff
@@ -207,17 +217,14 @@ public function getNextValidSibling($linkType, EntityInterface $previous)
...
         if (($index = $children->indexOf($following)) === false) {
             return null;
         }
 
-        for ($i = $index - 1; $i >= 0; --$i) {
-            $child = $children->get($i);
-            if ($child->hasCurrentRevision() && !$child->isTrashed()) {
-                return $child;
-            }
+        if($index -1 >=0){
```
Comment (inyono) [inline src/module/Entity/src/Entity/Entity/Entity.php]: space missing after `-` and  `>=`. (`$index-1 >= 0` would be okay for me, too)

File: src/module/Entity/src/Entity/Form/ModulePageForm.php [hunk 2]
```diff
@@ -32,21 +32,9 @@ function __construct(LicenseInterface $license)
...
-                'circle-question-mark' => 'Question mark (circle)',
-                'circle-exclamation-mark' => 'Exclamation mark (circle)',
-                'circle-info' => 'Info (circle)',
-                'pencil' => 'Pencil',
-                'file' => 'File',
-                'notes' => 'Notes',
-                'stats' => 'Stats',
-                'compass' => 'Compass',
-                'lightbulb' => 'Lightbulb'
+                '' => 'Explanation',
```
Comment (inyono) [inline src/module/Entity/src/Entity/Form/ModulePageForm.php]: Not sure how this is used in the front end. But if I understand this correctly, then `Explanation` means that no icon is used. So why not rename the option to `no icon`?

File: src/module/Ui/templates/entity/view/course-page.twig [hunk 3]
```diff
@@ -22,7 +22,7 @@
 
 <div class="row">
     <div class="col-xs-12">
-        <h2 class="heading-content">{{ revision.get('title') }}</h2>
+        <h1 class="heading-content">{{ revision.get('title') }}</h1>
```
Comment (arekkas) [inline src/module/Ui/templates/entity/view/course-page.twig]: there should only be one h1 per page which is usually the page header. is this the only h1 on that page?

File: src/module/Entity/src/Entity/Form/ModulePageForm.php (hunk 2 above)
Comment (Knorrke) [inline src/module/Entity/src/Entity/Form/ModulePageForm.php]: @inyono Yes, that's correct. Bit of a UX discussion here. When I talked with Michael about this, we wanted to make it clear to the user, when to use the "no icon" option, espacially destinguishing it from videos and exercises.

File: src/module/Ui/templates/entity/view/course-page.twig (hunk 3 above)
Comment (Knorrke) [inline src/module/Ui/templates/entity/view/course-page.twig]: @arekkas Yes, previously (when we still had the bootstrap carousel) there was the title of the course as h1, however in the current design only the title of the coursepage is displayed and therefore should be h1.

File: src/module/Entity/src/Entity/Form/ModulePageForm.php (hunk 2 above)
Comment (inyono) [inline src/module/Entity/src/Entity/Form/ModulePageForm.php]: ah, that's okay for me

**Comment by inyono** [comment]
LGTM

**Comment by arekkas** [comment]
LGTM @Knorrke

**Pull request merged**